modules/reitit-core/src/reitit/walk.clj

```diff
@@ -29,6 +29,14 @@
 (doseq [type [clojure.lang.PersistentArrayMap
               clojure.lang.PersistentHashMap
               clojure.lang.PersistentTreeMap]]
+  (extend type IKeywordize {:-keywordize -keywordize-kvreducible}))
+
+(defn- -keywordize-map
+  [m]
+  (let [f (fn [[k v]] (if (string? k) [(keyword k) v] [k v]))]
+    (into {} (map f) m)))
+
+(doseq [type [clojure.lang.IPersistentMap]]
```

Comment:
Why doseq here?

Reply:
Mostly because I copied the other implementation.
Makes it easier to extend to another type in the future